presto-main/src/main/java/com/facebook/presto/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -141,12 +147,28 @@ private Visitor(Session session)
         @Override
         public Map<PlanNodeId, SplitSource> visitTableScan(TableScanNode node, Void context)
         {
+            return visitScanAndFilter(node, Optional.empty());
+        }
+
+        private Map<PlanNodeId, SplitSource> visitScanAndFilter(TableScanNode scan, Optional<FilterNode> filter)
+        {
+            Set<DynamicFilter> dynamicFilters = filter
```

Comment:
Just a question.
Shouldn't dynamic filters be just pushed down to `TableScanNode`. And placed in `currentConstraint` field or some new field?

Reply:
We need to send information about dynamic filters to the split manager, so it can prune splits and in some cases also add a constraint to the Split (as Hive does for stripe pruning).